prototype/utils/arduino_serial_port.cpp

```diff
@@ -17,36 +17,44 @@ void setupSerial(const unsigned long baudRate) {
  * @param outArray The array in which to store the result of the read data.
  * @param sizeOfElement The size of each element in the array.
  */
-int readArray(const uint32_t arraySize, void* outArray,
-              const uint32_t sizeOfElement) {
-  readElement(outArray, arraySize * sizeOfElement);
+void readArray(const unsigned arraySize, void* outArray,
```

Comment:
I think we should either keep uint16_t or uin32_t in this function header. Otherwise code might work on some mcu but not on others as the size might change.

Reply:
Those types were never sent as serial data. They are only here to manage the data size to send and the size of each element. Therefore, them being unsigned will not change the behavior on other mcus.

Reply:
My point is that the current implementation limits the amount of data we can transfer through the serial interface. If we have a MCU with more than 32Kb of ram we won't be able to transfer the data since the const unsigned _might_ or _might not_ be big enough depending on the compiler and the parameters used. I think the size of the arraySize parameter should be uint32_t to make sure that it's big enough to handle everything.

Reply:
Arduino serial read only takes a `size_t`.  Using a `uint32_t` would be a misleading type. Should I use `size_t` instead of unsigned?